tests/ImageSharp.Tests/Image/ImageTests.WrapMemory.cs

```diff
@@ -12,12 +19,72 @@ public partial class ImageTests
     {
         public class WrapMemory
         {
+            class BitmapMemoryManager : MemoryManager<Bgra32>
+            {
+                private System.Drawing.Bitmap bitmap;
+
+                private BitmapData bmpData;
+
+                private int length;
+
+                public BitmapMemoryManager(Bitmap bitmap)
+                {
+                    if (bitmap.PixelFormat != PixelFormat.Format32bppArgb)
+                    {
+                        throw new ArgumentException("bitmap.PixelFormat != PixelFormat.Format32bppArgb", nameof(bitmap));
+                    }
+
+                    this.bitmap = bitmap;
+                    var rectangle = new Rectangle(0, 0, bitmap.Width, bitmap.Height);
+                    this.bmpData = bitmap.LockBits(rectangle, ImageLockMode.ReadWrite, bitmap.PixelFormat);
+                    this.length = bitmap.Width * bitmap.Height;
+                }
+
+                protected override void Dispose(bool disposing)
+                {
+                    this.bitmap.UnlockBits(this.bmpData);
+                }
+
+                public override unsafe Span<Bgra32> GetSpan()
+                {
+                    void* ptr = (void*) this.bmpData.Scan0;
+                    return new Span<Bgra32>(ptr, this.length);
+                }
+
+                public override unsafe MemoryHandle Pin(int elementIndex = 0)
+                {
+                    void* ptr = (void*)this.bmpData.Scan0;
+                    return new MemoryHandle(ptr);
+                }
+
+                public override void Unpin()
+                {
+                }
+            }
+
             [Fact]
-            public void ConsumedBuffer_IsMemoryOwner_ReturnsFalse()
+            public void WrapSystemDrawingBitmap()
             {
-                var memory = new Memory<int>(new int[55]);
-                var buffer = new ConsumedBuffer<int>(memory);
-                Assert.False(buffer.IsMemoryOwner);
+                using (var bmp = new Bitmap(51, 23))
+                {
+                    using (var memoryManager = new BitmapMemoryManager(bmp))
```

Comment:
👍 this is awesome..

Would love to be able to have a way for not needing to have the 2 `using`s here it would be nice if when constructing a wrapped `Image<TPixel>` we can tell it to own the memory manager directly and thus dispose of it when its disposed. That would then allow us to more easily develop an extension method directly against a `S.D.Image`.

Reply:
We need an `Image.WrapMemory<T>(IMemoryOwner<T> memoryOwner)` overload then. The [guideline tells](https://gist.github.com/GrabYourPitchforks/4c3e1935fd4d9fa2831dbfcab35dffc6#rule-8-if-you-have-an-imemoryownert-parameter-in-your-api-surface-you-are-accepting-ownership-of-that-instance) that the memory ownership will be transferred to `Image<T>` in that case, so it will be responsible for disposing the image.

Need to rethink the logic inside `BufferManager<T>` to correctly support this case in [`SwapOrCopyContent`](https://github.com/SixLabors/ImageSharp/blob/f9dbe41949921bd36fbb958dc7fb6a8c39f24280/src/ImageSharp/Memory/BufferManager.cs#L47).